doc/chips/10.rst

```diff
@@ -250,6 +250,78 @@ block while the rules for Phase 2 may be applied to the second, without a
 constant check to the location of the ``.init()`` call which is the linchpin for
 syntax 1.
 
+Child's fields could depend on parent's fields
+**********************************************
+
+We also considered allowing child's fields to be initialized based on
+parent's fields. For example, the child could control where, in its Phase 1,
+the parent's Phase 1 is invoked, not necessarily as the last step.
+
+The following example shows that based on the second syntax above.
+It also shows the option of having the child control when the parent's Phase 2
+is invoked.
+
+.. code-block:: chapel
+
+   class Parent {
+     var D: domain(1);
+     proc init(myRange) {
+       D = {myRange};  // Phase 1
+     } finalize {
+       ... Phase 2 code ...
+     }
+   }
+
+   class Child {
+     var xD: domain(1);
+     proc init(myRange, extra) {
+       super.init(myRange);   // initializes Parent.D
+       xD = D.expand(extra);  // initialize xD based on Parent.D
+     } finalize {
+       ... Phase 2 in Child ...
+       super.finalize();      // executes Phase 2 in Parent
+     }
+   }
+
+This approach seems to require the syntax 2.
+We chose not to incorporate it because it makes less clear
+how initialization occurs.
+
+Parent could complete initialization after child's initialization is complete
+*****************************************************************************
+
+We also considered allowing the parent class to perform final initialization
+actions after the child class(s) have established their initialization
+invariants. An example of such an action is adding the new object to a shared
+list. If the list is concurrently accessible, all other invariants on the
+new object must have been established prior to such addition.
+
+This can be done by introducing the optional "lastly" phase, to be executed
+in the parent after all actions specified by child class(s) have completed.
+It allows parent to assume "final control" over newly-created objects.
+
+The following example relies on syntax 2:
+
+.. code-block:: chapel
+
+   class Parent {
+     proc init(...) {
+       ....
+     } finalize {
+       writeln("Parent Phase 2");
+     } lastly {
+       AllObjects.add(this); // done after Parent's and Child's Phase 2
+     }
+   }
+   class Child: Parent {
+     proc init(...) finalize {
+       writeln("Child Phase 2");
+     }
+   }
+
+We did not incorporate this functionality because we preferred the user
+to perform such "final" actions outside of the initializer.
```

Comment:
Actually, I would prefer this section not be added at all.  I don't believe the group considered this a valid additional option, though others may disagree.